src/components/web3/SolLogin.tsx

```diff
@@ -0,0 +1,68 @@
+import { useEffect, useState } from "react";
+import { Trans, useTranslation } from "react-i18next";
+import { bodyText, walletAddress } from "./styles";
+import { web3LoginSol } from "./api";
+
+interface Props {
+  web3address?: string;
+  onAddressSelected: (address: string, event: string) => void;
+}
+
+export const SolLogin: React.FC<Props> = ({
+  web3address,
+  onAddressSelected,
+}) => {
+  const [notice, setNotice] = useState<JSX.Element | undefined>();
+  const { t } = useTranslation();
+
+  useEffect(() => {
+    window.braveSolana?.on("!!! accountChanged", () => setNotice(undefined));
+  }, []);
+
+  useEffect(() => {
+    if (!web3address) {
+      setNotice(<span>{t("wallet_connect_request")}</span>);
+      web3LoginSol()
+        .then((address) => {
+          onAddressSelected(address, "login");
+          setNotice(undefined);
+        })
+        .catch((err) => {
+          console.error("!!! web3 window.braveSolana.connect", err);
+          setNotice(
+            <Trans i18nKey="wallet_connect_failed">
+              <div>
+                <p>
+                  Failed to connect to wallet. Please reload the page and try
+                  again.
+                </p>
+                <a
+                  href="https://support.brave.com/hc/en-us/articles/4415497656461-Brave-Wallet-FAQ"
+                  style={{ color: "inherit" }}
+                >
+                  Help with Brave Wallet
```

Comment:
nit: Are we assuming users will only use brave wallet here to connect?

Reply:
The requirements that I know of:
- to create a room, you have to be using the Brave browser
- to create a web3 room, you need to have a wallet, though the Brave wallet is preferred ... all the Web3 Brave Talk documentation talks about only the Brave wallet, as a big part of web3 Brave Talk is to promote the  Brave wallet.

Reply:
> to create a web3 room, you need to have a wallet, though the Brave wallet is preferred ... all the Web3 Brave Talk documentation talks about only the Brave wallet, as a big part of web3 Brave Talk is to promote the Brave wallet.

My thinking is that users may want to utilize a different wallet if they're joining the call. Ultimately, this is a call by your team/product so non-blocking. I just wanted to call it out in case we were unintentionally causing friction the UX for non-brave wallet users. If it's intentional no concerns from me.

Reply:
I'm looking at the link -- https://brave.com/web3-talk-calls/ -- and it says ```Web3 calls are available to all Brave Talk users on any browser with any Web3 wallet. ``` which is what's on the web3 Brave Talk homepage.  is there some other/additional text that you'd recommend to that page?

@hadiamjad - i couldn't find the link that Kyle is referring to above. --  ttps://support.brave.com/hc/en-us/articles/4415497656461-Brave-Wallet-FAQ -- can you tell me how to find it? thanks!

Reply:
@mrose17  yea when Login fails or user does allow a selected account, the user is promoted with this link. You can find it `Login.tsx` (https://github.com/brave/brave-talk/blob/7aadf34ffe8dc13c2a3e6e85bfa6698d3e500c69/src/components/web3/Login.tsx#L37)and `SolLogin.tsx` (https://github.com/brave/brave-talk/blob/7aadf34ffe8dc13c2a3e6e85bfa6698d3e500c69/src/components/web3/SolLogin.tsx#L40)
<img width="750" alt="image" src="https://github.com/brave/brave-talk/assets/46374292/f0123c4a-527b-4365-8da6-fa64c7913397">

Reply:
Thanks. Well, that is a natural place to put the text and that link is where they need to go. The only other thing I think we could do is have another line saying "Not using Brave Wallet?" but I have no idea where that link should go. So, i don't think there is anything that can be done.

Reply:
I'd think an error message suggesting that user's contact their wallet's support channel would be useful. Otherwise a link to community channel would also be nice so that we can get reports of these failures and I can reach out to the appropriately wallet devs to help us be able to resolve the issue.

This is just a nit though, so that's up to you all to decide and I'm fine resolving with no action.